scripts/makeheader

```diff
@@ -3,9 +3,41 @@ from __future__ import print_function
 import re
 import os
 
-# expected format:
-# #define XF86XK_FooBar _EVDEVK(0x123) /* some optional comment */
-evdev_pattern = re.compile(r'^#define\s+XF86XK_(?P<name>\w+)\s+_EVDEVK\((?P<value>0x[0-9A-Fa-f]+)\)')
+# Expected format:
+#     #define XF86XK_FooBar 0x1234         /* some optional comment */
+# or:
+#     #define XF86XK_FooBar _EVDEVK(0x123) /* some optional comment */
+# We also need to match commented evdev entries:
+#     /* Use: XF86XK_FooBar _EVDEVK(0x123)    some optional comment */
+keysym_entry_pattern = re.compile(
+    r'''^
+    (?P<define>\#define|/\*\s+Use:)\s+
+    (?P<prefix>\w*)XK_(?P<name>\w+)(?P<spacing>\s+)
+    (?P<evdev>_EVDEVK\()?(?P<value>0x[0-9A-Fa-f]+)(?(evdev)\))
+    ''',
+    re.VERBOSE,
+)
+# Match remaining XK_ references
+xorgproto_keysym_prefix_pattern = re.compile(r'\b(?P<prefix>\w*)XK_(?!KOREAN\b)')
+
+
+def make_keysym_entry(m: re.Match[str]) -> str:
+    '''
+    Perform the substitutions
+    '''
+    if m.group('evdev'):
+        if m.group('define').startswith('#'):
+            # Replace the xorgproto _EVDEVK macro with the actual value
+            # 0x10081000 is the base, the evdev hex code is added to that.
+            # We replace to make parsing of the keys later easier.
+            value = 0x10081000 + int(m.group('value'), 16)
+            value_str = f'{value:#x}    '
+        else:
+            value_str = f'''_EVDEVK({m.group('value')})'''
+    else:
+        value_str = m.group('value')
+    return f'''{m.group('define')} XKB_KEY_{m.group('prefix') or ''}{m.group('name')}{m.group('spacing')}{value_str}'''
```

Comment:
I think this may be a bit more readable if we move do a `foo = m.group('foo')` and then use those variables instead. A few more lines but makes this particular format easier to read

Reply:
Done